### PR TITLE
Updated fav-icon Base URL from JupyterLab PageConfig.

### PR DESCRIPTION
**Pull request by jayeshsingh9767**
We were Getting 404 because base URL of binder(different base url) was used for fetching fav-icons I have update code to use Base URL from JupyterLab PageConfig. 
This closes issue #7076

### CODE REVIEW

File: packages/notebook-extension/src/index.ts [hunk 1]
```diff
@@ -422,10 +426,10 @@ const tabIcon: JupyterFrontEndPlugin<void> = {
       ) as HTMLLinkElement;
       switch (status) {
         case 'busy':
-          link.href = busyIcon;
+          link.href = baseURL + busyIcon;
```
Comment (jtpio) [inline packages/notebook-extension/src/index.ts:429]: Should this string concatenation be moved above where `notebookIcon` and `busyIcon` are defined?

Comment (jtpio) [reply packages/notebook-extension/src/index.ts:429]: Also it could be combined with `URLExt.join()` (some example uses in this repo).

Comment (jayeshsingh9767) [reply packages/notebook-extension/src/index.ts]: Thanks for the Quick review. This will be more readable. have Updated the PR.

**Comment by jayeshsingh9767** [comment]
@jtpio could u please guide me about why **Build / Test Lint (pull_request)** is failing?, Logs of the job are pointing to file [/notebook/blob/main/notebook/app.py](https://github.com/jupyter/notebook/blob/main/notebook/app.py).

**Comment by jtpio** [comment]
The lint failure looks related to an upstream change, maybe this new release? https://github.com/jupyter-server/jupyter_server/releases/tag/v2.8.0

Similar to https://github.com/jupyter/notebook/pull/7104, we can fix that separately.

**Comment by jtpio** [comment]
FYI @jayeshsingh9767 I've updated the branch to include https://github.com/jupyter/notebook/pull/7104, which should hopefully help fix the CI.

**Comment by jtpio** [comment]
The new failure can likely be fixed by running `jlpm prettier` at the top-level (normally done automatically via the `pre-commit` hook though).

File: packages/notebook-extension/src/index.ts [hunk 2]
```diff
@@ -38,6 +38,9 @@ import { Widget } from '@lumino/widgets';
 
 import { TrustedComponent } from './trusted';
 
+import { PageConfig, URLExt } from '@jupyterlab/coreutils';
```
Comment (jtpio) [inline packages/notebook-extension/src/index.ts:41]: These imports can be grouped with the other imports above.

Comment (jayeshsingh9767) [reply packages/notebook-extension/src/index.ts]: Got it will update the PR.

**Review by jtpio — APPROVED** [review]
Thanks!

Just checked with a different base URL and it seems to be working fine:


https://github.com/jupyter/notebook/assets/591645/c5ef8b77-ee95-449b-a3d3-246c1a374a47

**Comment by jayeshsingh9767** [comment]
Thanks @jtpio

**Pull request merged**